src/v/redpanda/admin_server.cc

```diff
@@ -3633,14 +3637,22 @@ ss::future<ss::json::json_return_type> admin_server::query_automated_recovery(
           "Unable to get controller leader, cannot get recovery status"};
     }
 
-    auto status = co_await _topic_recovery_status_frontend.local().status(
-      controller_leader.value());
-    if (!status) {
-        co_return ret;
+    auto extended = get_boolean_query_param(*req, "extended");
+    if (controller_leader.value() == config::node().node_id.value()) {
+        auto status_log = co_await _topic_recovery_service.invoke_on(
+          cloud_storage::topic_recovery_service::shard_id,
+          [](auto& svc) { return svc.recovery_status_log(); });
+        co_return serialize_topic_recovery_status(
+          cluster::map_log_to_response(std::move(status_log)), extended);
+    } else {
+        if (auto status = co_await _topic_recovery_status_frontend.local()
+                            .status(controller_leader.value());
+            status.has_value()) {
+            co_return serialize_topic_recovery_status(status.value(), extended);
+        }
     }
 
-    co_return serialize_topic_recovery_status(
-      status.value(), get_boolean_query_param(*req, "extended"));
+    co_return ret;
```

Comment:
nit: could transform this to 
```
...else if(...) { ... co_return serialize ... } 
else { co_return ret; }
```
to keep it in the same "level"

or remove the else and do directly 
``` 
if(...){ ... co_return serialize...}

co_return ret;
```

to keep it similar in structure with the rest of the function

Reply:
fixed